File: AuthenticationService.WinForms/Global/Session.cs
using AuthenticationService.Dtos.Authentication;
using AuthenticationService.Dtos.Users;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;

namespace AuthenticationService.WinForms.Global
{
    public static class Session
    {
        public static string? AccessToken { get; private set; }
        public static string? RefreshToken { get; private set; }

        public static DateTime AccessTokenExpiration { get; private set; }

        public static UserDetailsDto? User { get; private set; }

        public static void Start(LoginResponseDto loginResponse)
        {
            AccessToken = loginResponse.AccessToken;
            RefreshToken = loginResponse.RefreshToken;
            AccessTokenExpiration = loginResponse.AccessTokenExpiresAt;
        }

        public static void RefreshTokens(LoginResponseDto? response)
        {
            AccessToken = response?.AccessToken;
            RefreshToken = response?.RefreshToken;

            AccessTokenExpiration = response?.AccessTokenExpiresAt ?? default;
        }

        public static void Clear()
        {
            AccessToken = null;
            RefreshToken = null;

            AccessTokenExpiration = default;
        }
    }
}

File: AuthenticationService.Angular/src/app/models/authentication/session.ts
import { LoginResponseModel } from './login-response.model';
import { ProfileDetailsModel } from '../profile/profile-details.model';

export class Session {
  static AccessToken: string | null = null;

  static RefreshToken: string | null = null;

  static AccessTokenExpiration: Date | null = null;

  static User?: ProfileDetailsModel | null = null;

  static Start(loginResponse: LoginResponseModel): void {
    Session.AccessToken = loginResponse.accessToken;

    Session.RefreshToken = loginResponse.refreshToken;

    Session.AccessTokenExpiration = loginResponse.accessTokenExpiresAt;
  }

  static RefreshTokens(response: LoginResponseModel | null): void {
    Session.AccessToken = response?.accessToken ?? null;

    Session.RefreshToken = response?.refreshToken ?? null;

    Session.AccessTokenExpiration = response?.accessTokenExpiresAt ?? null;
  }

  static Clear(): void {
    Session.AccessToken = null;

    Session.RefreshToken = null;

    Session.AccessTokenExpiration = null;

    Session.User = null;
  }
}
